src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -145718,9 +145731,9 @@ partial class C
 ";
             var comp = CreateCompilation(src);
             comp.VerifyDiagnostics(
-                // (10,23): warning CS8819: Nullability of reference types in return type doesn't match partial method declaration.
+                // (10,23): warning CS8826: Partial method declarations 'T C.F<T>()' and 'T? C.F<T>()' have differences in parameter names, parameter types, or return types.
                 //     public partial T? F<T>() => default;
-                Diagnostic(ErrorCode.WRN_NullabilityMismatchInReturnTypeOnPartial, "F").WithLocation(10, 23)
+                Diagnostic(ErrorCode.WRN_PartialMethodTypeDifference, "F").WithArguments("T C.F<T>()", "T? C.F<T>()").WithLocation(10, 23)
```

Comment:
No idea if this is correct.

cc @cston this behavioral change is either coming directly from your previous PR #47576 or I might have done a mistake while fixing conflicts (Note: this test hasn't existed in #47576).

Also tagging @jcouv as this test was added in #49723.